Core2/Onboarding/src/main/java/com/infomaniak/library/onboarding/OnboardingScaffold.kt

```diff
@@ -0,0 +1,153 @@
+/*
+ * Infomaniak SwissTransfer - Android
+ * Copyright (C) 2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.library.onboarding
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.pager.HorizontalPager
+import androidx.compose.foundation.pager.PagerState
+import androidx.compose.foundation.pager.rememberPagerState
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.graphics.vector.ImageVector.Builder
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalLayoutDirection
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun OnboardingScaffold(
+    pagerState: PagerState,
+    onboardingPages: List<OnboardingPage>,
+    bottomContent: @Composable (PaddingValues) -> Unit,
+) {
+    Scaffold { paddingValues ->
+        Column {
+            val left = paddingValues.calculateLeftPadding(LocalLayoutDirection.current)
```

Comment:
The start/end and left/right thing did not seem like a very good approach, I managed to only use PaddingValues instances instead to solve this inconsistency.

The insets can't be set to the `Column` nor the `HorizontalPager` because I need the background part of the `OnboardingPageContent` to draw behind the system bars (at the top of the screen and horizontally). I also need the `HorizontalPager`'s content to draw inside the "safe areas" by default but have them drawn behind the system bars when they're scrolled (vertically or horizontally)